### PR TITLE
Adds local signing for metadata

### LINKED ISSUE #423: Task: Add option to sign a metadata without using API (local metadata)
### What is the task about?

Task: Add option to sign a metadata without using API (local metadata)

### Parent feature

_No response_

### References

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by enyinna1234**
This enables the use of a locally stored metadata to create a signature. It also saves the created signature locally.

Closes #423

# Description

Fixes #(issue)

# Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [x] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [ ] I agree to follow this project's Code of Conduct

### CODE REVIEW

File: tests/unit/cli/admin/test_metadata.py [hunk 1]
```diff
@@ -1194,3 +1194,30 @@ def test_metadata_sign_delete(self, client, test_context):
...
+            metadata.sign,
+            ["tests/files/das-root.json"],
+            input="\n".join(input_step),
+            obj=test_context,
+            catch_exceptions=False,
+        )
+
+        assert test_result.exit_code == 0, test_result.output
+        assert "Metadata Signed! 🔑" in test_result.output
+        assert "SIGNING KEYS" in test_result.output
```
Comment (kairoaraujo) [inline tests/unit/cli/admin/test_metadata.py:1222]: We've discussed this name in the table summary `SIGNING KEYS` is a bit confusing. Do you have a suggestion here @enyinna1234?

**Review by kairoaraujo — APPROVED** [review]

**Comment by kairoaraujo** [comment]
@enyinna1234 it will require a rebase

**Review by kairoaraujo — APPROVED** [review]

**Comment by kairoaraujo** [comment]
@enyinna1234 I spotted a required doc update as part of this PR.

https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/ff0360745e7670390ee89356181cea76378a3d9d/docs/source/guide/index.rst?plain=1#L676-L713

File: tests/unit/cli/admin/test_metadata.py [hunk 2]
```diff
@@ -1210,3 +1210,30 @@ def test_metadata_sign_delete(self, client, test_context):
...
+        fake_offline_md = {"metadata": json.loads(das_root)}
+        json.load = pretend.call_recorder(lambda *a: fake_offline_md)
+        json.dump = pretend.call_recorder(lambda *a: "fake_write")
+        test_result = client.invoke(
+            metadata.sign,
+            ["tests/files/das-root.json"],
+            input="\n".join(input_step),
+            obj=test_context,
+            catch_exceptions=False,
+        )
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_metadata.py:1234]: Make it more readable, separate function calls of a function you are testing from mocking.

```suggestion
        with open("tests/files/das-root.json", "r") as file:
            das_root = file.read()

        fake_offline_md = {"metadata": json.loads(das_root)}
        json.load = pretend.call_recorder(lambda *a: fake_offline_md)
        json.dump = pretend.call_recorder(lambda *a: "fake_write")

        test_result = client.invoke(
            metadata.sign,
            ["tests/files/das-root.json"],
            input="\n".join(input_step),
            obj=test_context,
            catch_exceptions=False,
        )
```

**Pull request closed without merging**